pkg/services/pluginsintegration/clientmiddleware/logger_middleware.go

```diff
@@ -10,28 +10,39 @@ import (
 	"github.com/grafana/grafana/pkg/plugins"
 	"github.com/grafana/grafana/pkg/plugins/instrumentationutils"
 	plog "github.com/grafana/grafana/pkg/plugins/log"
+	"github.com/grafana/grafana/pkg/plugins/manager/registry"
 	"github.com/grafana/grafana/pkg/plugins/pluginrequestmeta"
 )
 
 // NewLoggerMiddleware creates a new plugins.ClientMiddleware that will
 // log requests.
-func NewLoggerMiddleware(logger plog.Logger) plugins.ClientMiddleware {
+func NewLoggerMiddleware(logger plog.Logger, pluginRegistry registry.Service) plugins.ClientMiddleware {
```

Comment:
We have two different stores for plugins where one is reserved for "internal" use within `pkg/plugins` and the other is for all other parts of Grafana. You picked the internal one 😉 If you can replace this dependency with `pluginstore.Store` and expose Target as a field on `pluginstore.Plugin`, then that'd be swell 👍

Reply:
Yeah that would be ideal if you wouldn't mind 🙏

Reply:
I think this comment didn't get resolved? (maybe the commit was not properly pushed?)

Reply:
@svennergr spoke offline about it - turns out it's a bit trickier since it's used in other places I had overlooked. I'll have a look in a follow up but I think it may be not worth the effort 😞